### DISCUSSION TITLE
How to add new social networks

### DISCUSSION THREAD

**Discussion by Klaas-** [gohugo-ananke/ananke#805] [2 - Q&A / Support]
Hi,
I guess this is a question of "how to do it right".
I want to use wikipedia and facebook as a social link, wikipedia is not defined in the defaults of ananke ( https://github.com/theNewDynamic/gohugo-theme-ananke/blob/main/config/_default/params.toml ); facebook is.

Now if I try to add via my hugo.toml like this:
```
[params.ananke.social.follow]
networks = [
  "facebook",
  "wikipedia"
]

[params.ananke.social.facebook]
username = "facebook_username"

[params.ananke.social.wikipedia]
username = "wikipedia_article"

[[params.ananke.social.networks]]
slug = "wikipedia"
label = "Wikipedia"
color = "#0645ad"
profile = "https://en.wikipedia.org/wiki/%s"
icon = "wikipedia-w"
```

Now if I do that, only wikipedia works. I am guessing `[[params.ananke.social.networks]]` does not add a new element to that list, but rather overwrites it all? Is that on purpose or am I misunderstanding how the configuration merge with defaults is supposed to work?

My current working configuration is just copying the facebook definition from defaults:
```
[params.ananke.social.follow]
networks = [
  "facebook",
  "wikipedia"
]

[params.ananke.social.facebook]
username = "facebook_username"

[params.ananke.social.wikipedia]
username = "wikipedia_article"

[[params.ananke.social.networks]]
slug = "wikipedia"
label = "Wikipedia"
color = "#0645ad"
profile = "https://en.wikipedia.org/wiki/%s"
icon = "wikipedia-w"

[[params.ananke.social.networks]]
slug = "facebook"
label = "Facebook"
color = "#3b5998"
profile = "https://www.facebook.com/%s"
icon = "facebook" # font awesome brand icon name
link = "https://facebook.com/sharer/sharer.php"
[params.ananke.social.networks.particles]
u = "permalink"
```

it works.

So my question is did I misunderstand how this is supposed to work? Or should there maybe be a params.ananke.social.networks_default one that is merged with params.ananke.social.networks so that we only need to define additional socials in our own config?

Greetings
Klaas

**Comment by davidsneighbour** [comment]
You should spend 5 minutes to read through [this TOML introduction](https://learnxinyminutes.com/docs/toml/) :) I think your main issue is understanding how TOML "tables" work (about halfway down that page).

`[[something]]` is a "table row" meaning that `something` is extended by an additional array of data. So adding `[[params.ananke.social.networks]]` is extending the existing networks by a new one. 

And because I didn't see you mentioning the wiki, here is the link to the docs. I thought it is explaining pretty well [how to extend](https://github.com/theNewDynamic/gohugo-theme-ananke/wiki/Social-media-network-setup#setup-individual-new-networks). Copypaste the whole sample and exchange with your individual setup data. `slug`, `label`, `link` should be enough for Wikipedia.

**Comment by davidsneighbour** [comment]
Something along the lines of this, after the initial `ananke.social` setup:

```
[[ananke.social.networks]]
slug = "wikipedia"
label = "Wikipedia"
icon = "wikipedia" # font awesome brand icon name, didn't check if that is right
link = "https://wikipedia.com/"
```

**Comment by Klaas-** [comment]
I'll give you an example in a PR, that is not working, if I change the example site like this:
https://github.com/theNewDynamic/gohugo-theme-ananke/pull/806
https://deploy-preview-806--ananke-theme.netlify.app/


then I have a wikipedia logo, but none of the other defined ones, because -- to me it seems -- that setting https://github.com/theNewDynamic/gohugo-theme-ananke/blob/7bafcf27c05c1f57af39050ac0cb26de37563095/exampleSite/config.toml#L77-L82 in config.toml is overwriting the whole list, not extending the list

**Comment by davidsneighbour** [comment]
I wonder if that is an issue in the setup of the example site only. Let me go over it. in all my other projects having a single row will extend instead of override (in any TOML table). This might be related to the order of the variables. I'll add a working override example to the sample site if I can get it working.

**Comment by davidsneighbour** [comment]
This is a bug indeed. I am trying to fix it in the next day(s). 

https://github.com/theNewDynamic/gohugo-theme-ananke/issues/808

I will rework the way the networks are defined. That will need some testing and documentation rewrites.
